publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -321,6 +321,27 @@ public Entity migrate(Entity dataEntry) {
         return dataEntry;
     }
 
+    public List<FileEntry> fetchFileEntriesForResource(Resource resource) {
+        var partitionKeyValue = FileDao.getFileEntriesByResourceIdentifierPartitionKey(resource);
+
+        var queryRequest = new QueryRequest()
+                               .withTableName(tableName)
+                               .withIndexName(BY_TYPE_AND_IDENTIFIER_INDEX_NAME)
+                               .withKeyConditionExpression("PK3 = :resourceIdentifier AND begins_with(SK3, :type)")
+                               .withExpressionAttributeValues(
+                                   Map.of(
+                                       ":resourceIdentifier", new AttributeValue().withS(partitionKeyValue),
+                                       ":type", new AttributeValue().withS(FileDao.TYPE)
+                                   )
+                               );
```

Comment:
Extract method that tells us what this query does. If I understand it: `createQueryForFilesAssociatedWithResource`